eitprocessing/mixins/slicing.py

```diff
@@ -53,50 +42,55 @@ def __getitem__(self, key: slice | int):
             )
 
         if isinstance(key, int):
-            return self.select_by_index(start=key, end=key, end_inclusive=True)
+            return self.select_by_index(
+                start=key, end=key, start_inclusive=True, end_inclusive=True
+            )
 
-        raise TypeError(f"Invalid key type. Should be slice or int, not {type(key)}.")
+        raise TypeError(
+            f"Invalid slicing input. Should be `slice` or `int`, not {type(key)}."
+        )
 
     @abstractmethod
     def _sliced_copy(
         self, start_index: int, end_index: int, label: str | None = None
     ) -> Self:
-        ...
+        if label is None:
+            if start_index >= end_index:
+                pass
+            elif start_index < end_index - 1:
+                label = f"Frame ({start_index}) of <{self.label}>"
+            else:
+                label = f"Slice ({start_index}-{end_index-1}) of <{self.label}>"
 
-    def _check_time_sorted(self):
-        if not np.all(np.sort(self.time) == self.time):
-            raise ValueError(
-                f"Time stamps for {self} are not sorted and therefor data"
-                "cannot be selected by time."
-            )
+        ...
 
 
 class SelectByTime(SelectByIndex):
     def select_by_time(  # pylint: disable=too-many-arguments
         self,
         start: float | int | None = None,
         end: float | int | None = None,
-        start_inclusive: bool = True,
-        end_inclusive: bool = False,
         label: str | None = None,
```

Comment:
This forces a single behaviour. I'm not sure this is warranted with time
slicing.

With the way we chose to do time slicing, it is now not possible to slice a
sequence in parts, edit each part individually and then merge again, because
we chose to over-select when the time axes differ.
This is a use case I would like to consider. This requires the option to not
include the end.


I do agree it's best to have a default for index slicing, though.